test/functional/ui/cmdline_spec.lua

```diff
@@ -858,3 +860,156 @@ describe("cmdline height", function()
     assert_alive()
   end)
 end)
+
+describe('cmdheight can be 0', function()
+  local screen
+  before_each(function()
+    clear()
+    screen = Screen.new(25, 5)
+    screen:attach()
+  end)
+
+  it("cmdheight=1 noruler laststatus=2", function()
+    command("set cmdheight=1 noruler laststatus=2")
+    screen:expect{grid=[[
+      ^                         |
+      ~                        |
+      ~                        |
+      [No Name]                |
+                               |
+    ]]}
+  end)
+
+  it("cmdheight=0 noruler laststatus=2", function()
```

Comment:
```suggestion
  it("works with cmdheight=0 noruler laststatus=2", function()
```

Reply:
No need to revert, but for future reference "works with" is not necessary. It's just noise. The tests don't need to read like english sentences, that's a cornball feature of BDD.